abi-versions/v0.2.1/README.md

```diff
@@ -0,0 +1,1996 @@
+
+Data (of `data_size`) can be retrieved and/or modified using
+[`proxy_get_buffer_bytes`] and/or [`proxy_set_buffer_bytes`]
+with `buffer_id` set to `DOWNSTREAM_DATA`.
+
+Paused downstream can be resumed using [`proxy_continue_stream`]
+or closed using [`proxy_close_stream`] with `stream_type` set to
+`DOWNSTREAM`.
+
+Plugin must return one of the following values:
+- `CONTINUE` to forward `DOWNSTREAM_DATA` buffer upstream.
+- `PAUSE` to pause processing.
+
+
+#### `proxy_on_downstream_connection_close`
+
+* params:
+  - `i32 (uint32_t) stream_context_id`
+  - `i32 (`[`proxy_peer_type_t`]`) peer_type`
+* returns:
+  - none
+
+Called when downstream connection is closed.
+
+
+#### `proxy_on_upstream_data`
+
+* params:
+  - `i32 (uint32_t) stream_context_id`
+  - `i32 (size_t) data_size`
+  - `i32 (bool) end_of_stream`
+* returns:
+  - `i32 (`[`proxy_action_t`]`) action`
+
+Called for each data chunk received from upstream,
+even when the processing is paused.
+
+`data_size` represents total available size of the data that can be
+retrieved, and its value will increment if the processing is paused
+and data is buffered by the host and not forwarded downstream.
+
+Data (of `data_size`) can be retrieved and/or modified using
+[`proxy_get_buffer_bytes`] and/or [`proxy_set_buffer_bytes`]
+with `buffer_id` set to `UPSTREAM_DATA`.
+
+Paused upstream can be resumed using [`proxy_continue_stream`]
+or closed using [`proxy_close_stream`] with `stream_type` set to
+`UPSTREAM`.
+
+Plugin must return one of the following values:
+- `CONTINUE` to forward `UPSTREAM_DATA` buffer downstream.
+- `PAUSE` to pause processing.
+
+
+#### `proxy_on_upstream_connection_close`
+
+* params:
+  - `i32 (uint32_t) stream_context_id`
+  - `i32 (`[`proxy_peer_type_t`]`) peer_type`
+* returns:
+  - none
+
+Called when upstream connection is closed.
+
+
+## HTTP
+
+### Callbacks exposed by the Wasm module
+
+> **Note**
+> The same unique `stream_context_id` is shared by
+> HTTP request and response.
+
+#### `proxy_on_request_headers`
+
+* params:
+  - `i32 (uint32_t) stream_context_id`
+  - `i32 (size_t) num_headers`
+  - `i32 (bool) end_of_stream`
+* returns:
+  - `i32 (`[`proxy_action_t`]`) action`
+
+Called when HTTP request headers are received from downstream.
+
+All `num_headers` headers can be retrieved and/or modified using
+[`proxy_get_header_map_pairs`] and/or [`proxy_set_header_map_pairs`]
+with `map_id` set to `HTTP_REQUEST_HEADERS`.
+
+Individual HTTP request headers can be retrieved and/or modified using
+[`proxy_get_header_map_value`], [`proxy_replace_header_map_value`],
+[`proxy_add_header_map_value`] and/or [`proxy_remove_header_map_value`]
+with `map_id` set to `HTTP_REQUEST_HEADERS`.
+
+Paused request can be resumed using [`proxy_continue_stream`]
+or closed using [`proxy_close_stream`] with `stream_type` set to
+`HTTP_REQUEST`.
+
+Additionally, instead of forwarding request upstream, a HTTP response
+can be sent using [`proxy_send_local_response`].
```

Comment:
Seems like there are several variations of local responses like:
- sendDirectLocalReply https://github.com/envoyproxy/envoy/blob/50324229d7cf6024c43c7fd4bae1e042beea4728/source/common/http/filter_manager.cc#L1099
- sendLocalReply https://github.com/envoyproxy/envoy/blob/50324229d7cf6024c43c7fd4bae1e042beea4728/source/common/http/filter_manager.cc#L944
- sendLocalReplyViaFilterChain https://github.com/envoyproxy/envoy/blob/50324229d7cf6024c43c7fd4bae1e042beea4728/source/common/http/filter_manager.cc#L1061

Is it possible to make here `proxy_send_local_response` more explicitly to indicate which type specifically it will trigger or specifying clearly in the doc?